src/tasks/scenarios/project.ts

```diff
@@ -1,32 +1,77 @@
 import config = require('../../../config.js')
-import {cat, exec, mkdir, rm, touch, head} from 'shelljs'
-import { ProjectJob } from 'tasks/job'
+import { cat, exec, mkdir, rm, touch, head } from 'shelljs'
+import { ProjectJob } from '../jobs/project'
 import { ProjectResult } from 'types/result'
 import * as path from 'path'
-import * as fs from 'fs'
 import { Scenario } from 'tasks/scenario'
 import { download } from 'utils/request'
 
 export default class ProjectScenario extends Scenario {
   async setup(currentJobDir: string, job: ProjectJob) {
-    // TODO
-    const problemDir = path.join(currentJobDir, job.problem)
-    await download(job.problem, problemDir)
+    const problemZipDir = path.join(currentJobDir, 'problem.zip')
+    const solutionZipDir = path.join(currentJobDir, 'solution.zip')
+    await download(job.problem, problemZipDir)
+    await download(job.source, solutionZipDir)
 
-    const solutionDir = path.join(currentJobDir, job.source)
-    await download(job.problem, solutionDir)
+    const problemDir = path.join(currentJobDir, 'problem')
+    mkdir('-p', problemDir)
+    exec(`unzip ${problemZipDir} -d ${problemDir}`)
+
+    const solutionDir = path.join(currentJobDir, 'solution')
+    mkdir('-p', solutionDir)
+    exec(`unzip ${solutionZipDir} -d ${solutionDir}`)
+  }
+
+  run(currentJobDir: string, job: ProjectJob) {
+    const PROJECT_CONFIG = config.PROJECT[job.lang]
+    return exec(`docker run \\
+        --cpus="${PROJECT_CONFIG.CPU_SHARE}" \\
+        --memory="${PROJECT_CONFIG.MEM_LIMIT}" \\
+        --rm \\
+        -v "${currentJobDir}":/usr/src/runbox \\
+        -w /usr/src/runbox codingblocks/project-worker-"${job.lang}" \\
+        /bin/judge.sh -s "${job.submissionDirs}"
+    `);
   }
 
   async result(currentJobDir: string, job: ProjectJob): Promise<ProjectResult> {
-    // TODO
+    let result_code = cat(path.join(currentJobDir, 'result.code')).toString()
+    if (result_code) {
+      //problem hash and solution hash were not equal
+      return {
+        id: job.id,
+        stderr: cat(path.join(currentJobDir, 'result.stderr')).toString(),
+        stdout: '',
+        code: parseInt(result_code),
+        time: 0,
+        score: 0
+      }
+    }
+
+    // if hashes were equal, result_code will be found inside solution directory
```

Comment:
@jatinkatyal13 is this done on purpose? `result_code` and `result_time` inside solutionDirectory? and other stuff in the baseDirectory?

Reply:
Nice catch, I'll fix this [here](https://github.com/coding-blocks/judge-workers/blob/project-worker/containers/project-worker/base/judge.sh#L32). You can safely assume everything related to the result will be in the base directory